typha/pkg/snapcache/cache.go

```diff
@@ -298,32 +302,39 @@ func (c *Cache) fillBatchFromInputQueue(ctx context.Context) error {
 	}
 
 	log.Debug("Waiting for next input...")
-	select {
-	case obj := <-c.inputC:
-		log.WithField("update", obj).Debug("Got first update, peeking...")
-		storePendingUpdate(obj)
-	batchLoop:
-		for batchSize < c.config.MaxBatchSize {
-			select {
-			case obj = <-c.inputC:
-				storePendingUpdate(obj)
-			case <-ctx.Done():
-				log.WithError(ctx.Err()).Info("Context is done. Stopping.")
-				return ctx.Err()
-			default:
-				break batchLoop
+	for ctx.Err() == nil && !somethingToSend {
```

Comment:
This change cleans up a bunch of log spam in the FV tests (where we run with a very short wakeUpTicker).  It stops us from exiting to the calling function and getting its debug logs too.